src/lib_ccaligner/recognize_using_pocketsphinx.cpp

```diff
@@ -36,15 +36,15 @@ bool PocketsphinxAligner::processFiles()
     LOG("Audio Filename: %s Subtitle filename: %s", _audioFileName.c_str(), _subtitleFileName.c_str());
 
     std::cout << "Reading and processing subtitles...\n";
-    _subParserFactory = new SubtitleParserFactory(_subtitleFileName);
+    _subParserFactory = std::unique_ptr<SubtitleParserFactory>(new SubtitleParserFactory(_subtitleFileName));
```

Comment:
Why a unique pointer here?

Reply:
or it has to be raw pointer

Reply:
No, I'd say that it should just be a simple object, no need for heap allocations IMO.

Reply:
@Rakete1111 Note that it is not in constructor, and SubtitleParserFactory has no default constructor.